goi18n/merge_command.go

```diff
@@ -88,67 +83,81 @@ func (mc *mergeCommand) parse(arguments []string) {
 	sourceLanguage := flags.String("sourceLanguage", "en-us", "")
 	outdir := flags.String("outdir", ".", "")
 	format := flags.String("format", "json", "")
+	flat := flags.Bool("flat", true, "")
 
 	flags.Parse(arguments)
 
 	mc.translationFiles = flags.Args()
 	mc.sourceLanguage = *sourceLanguage
 	mc.outdir = *outdir
 	mc.format = *format
+	mc.flat = *flat
 }
 
 func (mc *mergeCommand) SetArgs(args []string) {
 	mc.translationFiles = args
 }
 
-type marshalFunc func(interface{}) ([]byte, error)
-
-func (mc *mergeCommand) writeFile(label string, translations []translation.Translation, localeID string, marshal marshalFunc) error {
+func (mc *mergeCommand) writeFile(label string, translations []translation.Translation, localeID string) error {
 	sort.Sort(translation.SortableByID(translations))
-	buf, err := marshal(marshalInterface(translations))
+
+	var iTranslations interface{}
```

Comment:
`mc.marshal(marshal(translations))` - it's also very strange.
I offer to name it as `conv2I` or just `convert`.